packages/oft-evm/contracts/OFTFeeAdapter.sol

```diff
@@ -12,59 +12,57 @@ abstract contract OFTFeeAdapter is OFTAdapter, Fee {
 
     uint256 public feeBalance;
 
-    constructor(address _token, address _lzEndpoint, address _delegate) OFTAdapter(_token, _lzEndpoint, _delegate) {}
+    error NoFeesToWithdraw();
+    event FeeWithdrawn(address indexed to, uint256 amountLD);
 
-    /**
-     * @dev Locks tokens from the sender's specified balance in this contract.
-     * @param _from The address to debit from.
-     * @param _amountLD The amount of tokens to send in local decimals.
-     * @param _minAmountLD The minimum amount to send in local decimals.
-     * @param _dstEid The destination chain ID.
-     * @return amountSentLD The amount sent in local decimals.
-     * @return amountReceivedLD The amount received in local decimals on the remote.
-     *
-     * @dev msg.sender will need to approve this _amountLD of tokens to be locked inside of the contract.
-     */
-    function _debit(
-        address _from,
-        uint256 _amountLD,
-        uint256 _minAmountLD,
-        uint32 _dstEid
-    ) internal virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
-        (amountSentLD, amountReceivedLD) = _debitView(_amountLD, _minAmountLD, _dstEid);
-        innerToken.safeTransferFrom(_from, address(this), amountSentLD);
-        feeBalance += (amountSentLD - amountReceivedLD);
+    constructor(
+        address _token,
+        address _lzEndpoint,
+        address _owner
+    ) OFTAdapter(_token, _lzEndpoint, _owner) {}
+
+    // @dev Fees accumulate inside of the contract to save gas, and then can be withdrawn by the owner.
+    function withdrawFees(address _to) external onlyOwner {
```

Comment:
Should we allow the ability to whitelist a non-owner to withdrawFees here?  I.e., `onlyWithdrawoooooor`?  Just seems like a multisig will own this, which is painful to use to withdraw fees.